crates/core/src/config.rs

```diff
@@ -84,25 +84,37 @@ impl Default for Config {
 }
 
 impl Config {
-    pub fn from_env() -> Self {
-        Self {
+    pub fn from_env() -> Result<Self> {
+        fn require_env_var(var_key: &'static str) -> Result<String> {
+            std::env::var(var_key).map_err(|_| eyre!("The \"{}\" env var must be set or a configuration file must be specified", var_key))
+        }
+
+        Ok(Self {
             network: Network::from_str(
                 &std::env::var("NETWORK").unwrap_or_default(),
             )
             .unwrap_or(Network::MAINNET),
-            eth_execution_rpc: std::env::var("ETH_EXECUTION_RPC")
-                .unwrap_or_default(),
-            starknet_rpc: std::env::var("STARKNET_RPC").unwrap_or_default(),
+            eth_execution_rpc: require_env_var("ETH_EXECUTION_RPC")?,
```

Comment:
Extra sanity checks for RPC URLs might be useful here, just to fail fast if invalid config param was provided

Reply:
Yes, we probably could also change the type of the *_RPC vars to proper `Url`s instead of `String`s.

What do you think about me handling this in a separate issue?
Also I'm bothered with the *_RPC suffix, what do you think about changing this to *_URL?

Reply:
I think that you shouldn't remove the `_rpc` suffix as it tells what it really is, but at the same time I think that adding `_url` to it would be beneficial too, so in the end you would end up in something like `eth_execution_rpc_url`.

Reply:
Both params are passed as strings later on if I'm not mistaken, so I think some basic URL validation seems reasonable here.

I personally don't care much about naming and suffixes, but any changes there require respective update of readme and/or other docs to avoid users wondering.